### DISCUSSION TITLE
Is there a way that I can use rotating while fading?

### DISCUSSION THREAD

**Discussion by Stanford997** [ManimCommunity/manim#3263] [Q&A] [answered]
Here is my code:
```python
        text = Text("rotate while fading", height=1)
        self.add(text)
        self.play(
            Rotating(
                text,
                PI / 2,
                run_time=3,
                about_edge=(np.array([0, 1, 0])),
                axis=LEFT),
            FadeOut(text)
        )
``` 
But it doesn't work. It just fades out without rotating.

**Comment by uwezi** [comment, marked as answer]
Ok, let's first get your rotation to work at all, currently this part alone would cause a couple of error messages:

```py
class rotfade(Scene):
    def construct(self):
        text = Text("rotate while fading", height=1)
        self.add(text)
        self.play(
            Rotating(
                text,
                angle=PI / 2,
                run_time=3,
                about_edge=np.array([0, 1, 0]),
                axis=LEFT,
                ),
        )  
```

https://github.com/ManimCommunity/manim/assets/8582807/5e5746b2-f5b3-41f8-99a4-babf43b1814b



You cannot have two independent animations on the same object in a single `self.play()` because of the way the rendering engine in Manin works, but you can do it like this, using basic object modifier together with `.animate`:

```py
class rotfade2(Scene):
    def construct(self):
        text = Text("rotate while fading", height=1)
        self.add(text)
        self.play(
            text.animate.rotate(angle=PI/2,about_point=text.get_top(), axis=LEFT).set_opacity(0)
        )  
```

https://github.com/ManimCommunity/manim/assets/8582807/84f383ff-115a-482a-a6b5-2ebe32f9fdee

Come over to Discord to learn more about Manim and get better help:
[https://docs.manim.community/en/stable/faq/general.html](https://docs.manim.community/en/stable/faq/general.html?highlight=discord#where-can-i-find-more-resources-for-learning-manim)

**Reply by Stanford997** [reply]
Thanks to your help I managed to achieve this:
```python
class rotfade3(Scene):
    def construct(self):
        text1 = Text("rotate while fading", height=1)
        self.add(text1)

        text2 = Text("fade while rotating", height=1)
        self.add(text2)

        text2.rotate(angle=3 * PI / 2, about_point=[0, 0, -0.5], axis=LEFT).set_opacity(0)
        self.play(
            text1.animate.rotate(angle=PI / 2, about_point=[0, 0, -0.5], axis=LEFT).set_opacity(0),
            text2.animate.rotate(angle=PI / 2, about_point=[0, 0, -0.5], axis=LEFT).set_opacity(1)
        )
``` 

https://github.com/ManimCommunity/manim/assets/52944194/eb19dca0-0f02-4130-b1cf-228a7d42d4af

**Reply by uwezi** [reply]
nice!
